ingest/sqs/service/sqs_streaming_service.go

```
@@ -111,19 +113,49 @@ func (s *sqsStreamingService) processBlockRecoverError(ctx sdk.Context) (err err
 			// so that the next block processes only the pools that were changed.
 			s.blockProcessStrategyManager.MarkInitialDataIngested()
 		}
-	}()
 
-	blockProcessor := blockprocessor.NewBlockProcessor(s.blockProcessStrategyManager, s.grpcClient, s.poolsExtractor, s.poolsTransformer, s.nodeStatusChecker, s.blockUpdatesProcessUtil)
+		errCh <- err
+	}
+
+	var (
+		wg    sync.WaitGroup
+		errCh = make(chan error, len(s.grpcClient) * 2)
```

golangci-lint (1.62.2)

File is not `goimports`-ed

(goimports)